feat: rewrite Admin-SM API auth guide (#4215)
camunda/camunda-docs: docs/apis-tools/administration-sm-api/administration-sm-api-authentication.md

description: "The Administration API for Self-Managed is a REST API and provides access to Console Self-Managed data. Requests and responses are in JSON notation."
---

> All Administration Self-Managed API requests require authentication. To authenticate, generate a [JSON Web Token (JWT)](https://jwt.io/introduction/) and pass it in each request.

Reviewer: ```suggestion
All Administration Self-Managed API requests require authentication. To authenticate, generate a [JSON Web Token (JWT)](https://jwt.io/introduction/) and include it in each request.
```

> ## Generating a token

Reviewer: ```suggestion
## Generate a token
```
I know we use gerunds (ing) but not sure we should? One to think about, perhaps not for this PR

Author: I'm going to hold off on this one for now.

2. [Add permissions to this application](/self-managed/identity/user-guide/additional-features/incorporate-applications.md) for **Administration Self-Managed API**.
3. Capture the `Client ID` and `Client Secret` from the application in Identity.
4. [Generate a token](/self-managed/identity/user-guide/authorizations/generating-m2m-tokens.md) to access the REST API. Provide the `client_id` and `client_secret` from the values you previously captured in Identity.
   ```shell
   curl --location --request POST 'http://localhost:18080/auth/realms/camunda-platform/protocol/openid-connect/token' \
   --header 'Content-Type: application/x-www-form-urlencoded' \
   --data-urlencode "client_id=${CLIENT_ID}" \
   --data-urlencode "client_secret=${CLIENT_SECRET}" \
   --data-urlencode 'grant_type=client_credentials'
   ```
5. A successful authentication response looks like the following:
   ```json
   {
     "access_token": "<TOKEN>",
     "expires_in": 300,
     "refresh_expires_in": 0,
     "token_type": "Bearer",
     "not-before-policy": 0
   }
   ```
6. Capture the value of the `access_token` property and store it as your token.

## Using a token

> Include the captured token as an authorization header in each request: `Authorization: Bearer <TOKEN>`.

Reviewer: ```suggestion
Include the previously captured token as an authorization header in each request: `Authorization: Bearer <TOKEN>`.
```
Suggestion

> For example, to call the Administration Self-Managed API's ["Get current clusters" endpoint](./specifications/get-clusters.api.mdx), send the following request against the target environment:

Reviewer: ```suggestion
For example, to send a request to the ["Get current clusters" endpoint](./specifications/get-clusters.api.mdx):
```
Edited for conciseness, not sure if the target environment needs stating though?

:::tip

> The `${CAMUNDA_BASE_URL}` variable below represents the URL of the Self-Managed environment. You can configure this value in your Self-Managed installation. The default value is `http://localhost:8080`.

Author: @urbanisierung there are a few things here I'm hoping you can clarify for me: 

1. I don't like how I described this: "the URL of the Self-Managed environment." Which component is it associated with? Zeebe?

2. Is 8080 the default port in a self-managed environment? When I spun up an 8.6-alpha4 environment via Docker Compose from the camunda/camunda-platform repo, the port associated with Zeebe was 8088. I'm not sure what other installation methods use as a default. 

3. I usually try to test these auth guides against a real environment, to make sure the scripts work when copy-pasted. With my 8.6-alpha4 environment from Docker Compose, I got a 404 error for the path `/admin-api/clusters`, against any of the running ports on `localhost`. Is there something I need to do to enable this API when setting up my environment?

Reviewer 2: Hey @pepopowitz! Sorry for the late response, I was on FTO.

> I don't like how I described this: "the URL of the Self-Managed environment." Which component is it associated with? Zeebe?

This is all about the admin API of console self-managed. Basically a very first iteration with some basic capabilities as described here: https://github.com/camunda-cloud/camunda-cloud-management-apps/issues/2871

> Is 8080 the default port in a self-managed environment?

Yesyes

> When I spun up an 8.6-alpha4 environment via Docker Compose from the camunda/camunda-platform repo, the port associated with Zeebe was 8088. I'm not sure what other installation methods use as a default.

There is currently no way to use console self-managed with docker compose. The console app relies on information provided by Helm Charts.

> I usually try to test these auth guides against a real environment, to make sure the scripts work when copy-pasted. With my 8.6-alpha4 environment from Docker Compose, I got a 404 error for the path /admin-api/clusters, against any of the running ports on localhost. Is there something I need to do to enable this API when setting up my environment?

The Admin API is enabled for versions starting with `8.6.0-alpha4`. But since there is no docker compose flow, the console pod won't boot up. If you need a quick'n'dirty way to use helm: I've created a very basic (and most probably buggy xD ) way to spawn up a self-managed installation with helm: https://github.com/urbanisierung/csctl.

> The `${CAMUNDA_BASE_URL}` variable below represents the URL of the Self-Managed environment. You can configure this value in your Self-Managed installation. The default value is `http://localhost:8080`.

Reviewer: I'm not sure if this tip is better below the code sample, as the intro paragraph suggests "To send a..." but then this tip interrupts what I am expecting.

Author: Also not sure, going to leave this one for now.

:::

```shell
curl --request GET ${CAMUNDA_BASE_URL}/admin-api/clusters \
   --header "Authorization: Bearer ${TOKEN}"
```

> A successful response would include [cluster information](./specifications/get-clusters.api.mdx). For example:

Reviewer: ```suggestion
A successful response includes [cluster information](./specifications/get-clusters.api.mdx). For example:
```
Edited for conciseness
